clar2wasm/src/words/equal.rs

```diff
@@ -403,9 +403,34 @@ fn wasm_equal(
         }
         // is-eq-bytes function can be used for types with (offset, length)
         TypeSignature::SequenceType(SequenceSubtype::BufferType(_))
-        | TypeSignature::SequenceType(SequenceSubtype::StringType(StringSubtype::ASCII(_)))
-        | TypeSignature::SequenceType(SequenceSubtype::StringType(StringSubtype::UTF8(_)))
-        | TypeSignature::PrincipalType
+        | TypeSignature::SequenceType(SequenceSubtype::StringType(_)) => {
+            if matches!(
+                (ty, nth_ty),
+                (
+                    TypeSignature::SequenceType(SequenceSubtype::BufferType(_)),
+                    TypeSignature::SequenceType(SequenceSubtype::BufferType(_)),
+                ) | (
+                    TypeSignature::SequenceType(SequenceSubtype::StringType(StringSubtype::ASCII(
+                        _
+                    ))),
+                    TypeSignature::SequenceType(SequenceSubtype::StringType(StringSubtype::ASCII(
+                        _
+                    ))),
+                ) | (
+                    TypeSignature::SequenceType(SequenceSubtype::StringType(StringSubtype::UTF8(
+                        _
+                    ))),
+                    TypeSignature::SequenceType(SequenceSubtype::StringType(StringSubtype::UTF8(
+                        _
+                    ))),
+                )
+            ) {
+                wasm_equal_bytes(generator, builder, first_op, nth_op)
+            } else {
+                no_type_match()
+            }
```

Comment:
Sorry, I wasn’t clear enough yesterday. I meant that this portion should be a `match` statement instead of an if else on the result of a `match` statement.

Reply:
clippy does not approve of this match, since it can be done with matches! macro

Reply:
Oops, my bad then.